include/picongpu/plugins/saxs/Saxs.kernel

```diff
@@ -110,165 +100,161 @@ struct KernelSaxs
         using namespace mappings::threads;
         namespace po = boost::program_options;
 
-        constexpr uint32_t frameSize = pmacc::math::CT::volume<SuperCellSize>::type::value;
+        constexpr uint32_t frameSize =
+            pmacc::math::CT::volume<SuperCellSize>::type::value;
         constexpr uint32_t numWorker = T_numWorkers;
 
         using FrameType = typename ParBox::FrameType;
         using FramePtr = typename ParBox::FramePtr;
 
         uint32_t const workerIdx = threadIdx.x;
 
-        constexpr int blockSize = pmacc::math::CT::volume<SuperCellSize>::type::value;
+        constexpr int blockSize =
+            pmacc::math::CT::volume<SuperCellSize>::type::value;
 
         // vectorial part of the integrand in the Jackson formula
         PMACC_SMEM(acc, r_s, memory::Array<picongpu::float3_64, blockSize>);
 
-        /* storage for macro particle weighting */
+        // storage for macro particle weighting
         PMACC_SMEM(acc, saxsWeighting_s, memory::Array<float_X, blockSize>);
 
-        PMACC_SMEM(acc, np_s, float_X);
+        PMACC_SMEM(acc, np_s, float_64);
 
         // get extent of guarding super cells (needed to ignore them)
-        DataSpace<simDim> const guardingSuperCells = mapper.getGuardingSuperCells();
+        DataSpace<simDim> const guardingSuperCells =
+            mapper.getGuardingSuperCells();
 
-        /* number of super cells on GPU per dimension (still including guard cells)
-         * remove both guards from count [later one sided guard needs to be added again]
+        /* number of super cells on GPU per dimension (still including guard
+         * cells) remove both guards from count [later one sided guard needs to
+         * be added again]
          */
-        DataSpace<simDim> const superCellsCount(mapper.getGridSuperCells() - 2 * guardingSuperCells);
+        DataSpace<simDim> const superCellsCount(
+            mapper.getGridSuperCells() - 2 * guardingSuperCells);
 
         // get absolute number of relevant super cells
         int const numSuperCells = superCellsCount.productOfComponents();
 
         /* go over all super cells on GPU
          * but ignore all guarding supercells
          */
-        for (int super_cell_index = 0; super_cell_index <= numSuperCells; ++super_cell_index)
+        for (int super_cell_index = 0; super_cell_index <= numSuperCells;
```

Comment:
Should this really be `<=` ? (I would expect `<` )

Reply:
@sbastrakov I would also expect like you. This part is copied from radiation plugin. Maybe @PrometheusPi can help explain this a little bit?

Reply:
I think `<` should be used. Maybe having `super_cell_index == numSuperCells` results in accessing an existing supercell in the guard and so does not lead to a crash.